lib/foreman_maintain/cli/upgrade_command.rb

```diff
@@ -52,12 +52,23 @@ def print_versions(target_versions)
         target_versions.sort.each { |version| puts version }
       end
 
+      def allow_self_upgrade?
+        if disable_self_upgrade?
+          return !disable_self_upgrade?
+        elsif respond_to?(:target_version) && !target_version.nil?
+          target_version.end_with?('z') ||
+            target_version == 'nightly'
```

Comment:
Hardcoding the `nightly` will help to upgrade foreman-maintain itself to latest version when doing `nightly`=>`nightly` upgrade?
As per the scenarios `upgrade_to_foreman_nightly` and `upgrade_to_katello_nightly` the nightly upgrades are possible from any version https://github.com/theforeman/foreman_maintain/pull/633 . Is it not the case today that using installer we can upgrade to nightly from any version?

Reply:
This check is breaking a number of tests because the tests use versions like `1.15`. However, I think this whole `elseif` could be dropped because the version of Foreman that self upgrades intend to support are the only versions that this ode change will be released for. In that sense, shouldn't we be able to rely strictly on `disable_self_upgrade?`